signin_parser_spec.rb

```diff
@@ -12,26 +12,30 @@
   end
 
   it "should be signed in and sign out" do
-    signed = @signin_parse.sign_in(@url, @login, @password, @enter)
+    signed = @signin_parse.sign_in(@url, @login, @password, @enter, @quit)
     expect(signed.class).to eq Capybara::Session
     expect(signed.find('a', text: @quit).class).to eq Capybara::Node::Element
     @signin_parse.sign_out(@quit)
     expect(signed.find_button(@enter).class).to eq Capybara::Node::Element
   end
 
-  it "should be valid login and password" do
-    signed = @signin_parse.sign_in(@url, '', '', @enter)
-    begin
-      quit_found = signed.find('a', text: @quit)
-    rescue Capybara::ElementNotFound
-      quit_found = false
-    end
-    expect(quit_found.class).to eq FalseClass
+  it "should be valid url, login and password" do
+    bad_url = "abra-kad.abra"
+    signed = @signin_parse.sign_in(bad_url, '', '', @enter, @quit)
+    expect(signed.class).to eq String
+    expect(signed).to eq "URL #{bad_url} is invalid"
+    visit_error_url = "http://abrakadabra.vladloz.pp.ua"
+    signed = @signin_parse.sign_in(visit_error_url, '', '', @enter, @quit)
+    expect(signed.class).to eq String
+    expect(signed).to eq "Visit error for #{visit_error_url}"
+    signed = @signin_parse.sign_in(@url, '', '', @enter, @quit)
+    expect(signed.class).to eq String
+    expect(signed).to eq "Login/password combination is invalid"
   end
```

Comment:
checkout before{} and each{} blocks in rspecs, general idea, that in "it" you should be only testing something, not testing and prep some data for other test. Also tests should be separated, so instead of one big test with everything, you need to have many small ones. That way specs are much more usable and give you exact place to look in code once they fail.